### PR TITLE
Make font drawing more generic

### PR DESCRIPTION
**Pull request by amylizzle**
gonna steal this for OD's maptext implementation

it'd be helpful if I could draw it to `DrawingWorldHandle`

### CODE REVIEW

**Comment by PJB3005** [comment]
Does this actually work? Screen and world have inverted coordinate spaces, I can't imagine how the font renderer would just work with minimal changes like this.

**Comment by amylizzle** [comment]
Yup 
image

**Comment by PJB3005** [comment]
Uh, what happens if you try to render text with a period? That `!` looks like it's positioned wrong...

**Comment by amylizzle** [comment]
image

Yeah it's a little offset from where it ought to be. I did have to apply the character metric's offset again in OD's content-side code.

File: Robust.Client/Graphics/Font.cs [hunk 1]
```diff
@@ -132,7 +132,10 @@ public override float DrawChar(DrawingHandleScreen handle, Rune rune, Vector2 ba
             }
 
             baseline += new Vector2(metrics.Value.BearingX, -metrics.Value.BearingY);
-            handle.DrawTexture(texture, baseline, color);
+            if(handle is DrawingHandleWorld worldhandle)
+                worldhandle.DrawTextureRect(texture,  Box2.FromDimensions(baseline, texture.Size));
+            else
+                handle.DrawTexture(texture, baseline, color);
```
Comment (PJB3005) [inline Robust.Client/Graphics/Font.cs:138]: Really not a fan of this type check given that this is hot path code.

Comment (amylizzle) [reply Robust.Client/Graphics/Font.cs:138]: best alternative I can think of is to split `DrawChar()` into two function signatures, one with `DrawingHandleWorld` and one with `DrawingHandleScreen`. Bit code-duplicate-y but shouldn't be a slowdown.

**Pull request merged**